text/0021-internal-data-monitoring.md

```diff
@@ -0,0 +1,182 @@
+method of displaying data (bar graph, pie chart, etc) and choosing a
+specific time period with various filters.
+
+### Project 2: Social Media Data Vis
+The second project would be a program that monitors how others interact
+with Creative Labs. This would be primarily to understand the correlation
+between social media posts interactions (ads, likes, shares) and event
+attendance. 
+
+### Project 3: Website Data Vis
+The third project would dive into where (and at what frequency) users
+are navigating towards on our main website.
+
+#### Future Goals
+The obvious first direction for future goals after these projects is
+for new data analysis projects to be added and implemented into this
+monorepo. It would be beneficial for there to be a system in place where
+anyone on Creative Labs board can submit an issue in regards to CL data
+- whether it's displaying a certain dataset or exploring if there is
+a correlation between two datasets - that the dev team can then add to
+existing projects or start a new project if necessary.
+
+## Why are we doing this?
+
+Event attendance and participation is frequently stressed within internal
+board goals, and it would be great to see how we can collectively improve
+those numbers and grow our organization! Watchdog allows us to implement
+a system and benefit from years of past datasets that can help us make
+informed decisions in the future. This would be especially great for
+design and marketing teams so that they can focus on strategies that
+have been proven to work.
+
+## What is the high level design/implementation?
+
+The first project on project data would require a web app that allows board
+members to authenticate themselves as the data is sensitive and for
+internal use only as of now. Following the authentication there needs to
+be the development of the web app that displays the most important parts
+of the dataset. After this, options should be implemented for the user
+to be able to select specific project types or time periods. The second
+project into social media data and third project for main website data
+would be essentially the same structure. The difference would be that the
+data is from different sources - split up by each notable Creative Labs event
+(Demo Day or General Meeting) for Project 2 and split by portions of the
+main website for Project 3.
+
+Main parts of Project 1:
+* displaying important comparisons of data
+* options for user selection
+    * time period
+    * project type
+    * member major
+    * member year
+
+We could use the following data tables for each member application and time period.
+
+Member application
+```json
+{
+    "<name>": {
+        "year": "<year>",
+        "major": "<major>",
+        "email": "<email>",
+        "timestamp": "<timestamp>",
+        "project_choice_order": "[<choice_1>, <choice_2>, ...]",
+        "other_application_cycles": "[<cycle_1>, <cycle_2>, ...]"
```

Comment:
not sure if this works? you are gonna put what time it is but how does that relate to the proejct's they applied for?

Reply:
i wanted to find if theres a correlation between how early/late they submitted their application with if they were accepted into their projects

Reply:
maybe a better model is to have applications as a separate data model? And have a project member get linked to an application and vice versa. Think about how you will query all this info for the frontend! What is the easiest way for you to accomplish the query using best fitted data model?

Reply:
^these questions are why we do RFC! you go through the hard work upfront to understand the problem at a high level, so that when you start building it, you can easily make decisions that wont lead you astray